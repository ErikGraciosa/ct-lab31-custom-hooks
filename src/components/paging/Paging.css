.Paging {
  font-size: 20pt;
}

.dark {
  background-color: black;
  color: white;
}

.light {
  background-color: white;
  color: black;
}
