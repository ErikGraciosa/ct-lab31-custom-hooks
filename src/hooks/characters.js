import { useState, useEffect } from 'react';
import { getAllCharacters, getDetails } from '../services/rickAndMortyApi';

export const useCharacters = () => {
  const [loading, setLoading] = useState(true); //initialize loading to true
  const [quotes, setQuotes] = useState([]); //initialize quotes to empty string

  useEffect(() => {
    getAllCharacters()
      .then((quotes) => {
        setQuotes(quotes.results);
        setLoading(false);
      });
  }, []);

  return {
    loading,
    quotes
  };
};

export const useDetails = (id) => {
  const [loading, setLoading] = useState(true);
  const [info, setInfo] = useState([]);

  useEffect(() => {
    getDetails(id)
      .then((info) => {
        setInfo(info);
        setLoading(false);
      });
  }, []);

  return {
    loading,
    info
  };

};
