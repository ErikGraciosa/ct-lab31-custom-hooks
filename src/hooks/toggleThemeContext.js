import React, { createContext, useState, useContext } from 'react'; 
import PropTypes from 'prop-types';

const ThemeContext = createContext(null);

export const ThemeProvider = ({ children }) => {
  const [theme, setTheme] = useState('light');

  return (
    <ThemeContext.Provider value={{ theme, setTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};

ThemeProvider.propTypes = {
  children: PropTypes.any
};

export const useTheme = () => {
  const { theme, setTheme } = useContext(ThemeContext);
  return { 
    theme,
    setTheme
  };
};
