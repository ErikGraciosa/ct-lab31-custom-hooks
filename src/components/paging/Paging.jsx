import React from 'react';
import PropTypes from 'prop-types';
import styles from './Paging.css';
import { useTheme } from '../../hooks/toggleThemeContext';

const Paging = ({ increment, decrement, count }) => {
  const { theme } = useTheme(); //this needs to be what toggles
  console.log(theme);
  return (
    <div>
      <button onClick={decrement}>-1</button>
      <span>--{count}--</span>
      <button onClick={increment}>+1</button>
      <div className={`${styles.Paging} ${styles[theme]}`}>Theme Color</div>
    </div>
  );
};

Paging.propTypes = {
  increment: PropTypes.func,
  decrement: PropTypes.func,
  count: PropTypes.number.isRequired
};

export default Paging;
