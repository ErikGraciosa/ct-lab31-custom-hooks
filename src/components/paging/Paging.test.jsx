import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Paging from './Paging';
import { ThemeProvider } from '../../hooks/toggleThemeContext';
import { MemoryRouter } from 'react-router-dom';

describe('Paging component', () => {
  afterEach(() => cleanup());
  it('renders Paging', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <MemoryRouter>
          <Paging />
        </MemoryRouter>     
      </ThemeProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
