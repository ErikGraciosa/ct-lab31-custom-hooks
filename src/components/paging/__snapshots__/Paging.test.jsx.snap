// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paging component renders Paging 1`] = `
<DocumentFragment>
  <div
    class="light"
  >
    <button>
      -1
    </button>
    <span>
      ----
    </span>
    <button>
      +1
    </button>
  </div>
</DocumentFragment>
`;
